.github/workflows/conda.yml

```diff
@@ -105,7 +105,7 @@ jobs:
             export EXTRA_OPTIONS=" "
           fi
 
-          python -m pytest -n auto --dist loadscope $EXTRA_OPTIONS tests
+          python -m pytest -n 4 --dist loadscope $EXTRA_OPTIONS tests
```

Comment:
Interesting to see that `auto` is only giving us 2.

Reply:
We probably only have 2 cores, but almost all of our computation is happening remotely.  The platform cranks this up pretty high, into the 10s I think.

Reply:
Yeah, `auto` giving us 2 means we have 2 cores on the GHA runners. Parallelizing has gotten tests (just the `pytest` part of the full CI build) down to ~5 minutes compared to ~12 minutes on `main` :tada:. Bumping the number of `pytest-xdist` workers from 2 to 4 didn't change the overall test runtime because we're currently bottlenecked by the h2o benchmarks (which is the module that takes the longest to run at ~5 minutes). I'm going to bump to `-n 10` to make things a bit more future-proof as we add more test modules, but otherwise I think this is good to go